src/engine/input/key_mapping/InputDataList.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+
+/// <summary>
+///   Stores an godot input action and its associated events
+/// </summary>
+public class InputDataList : ICloneable
```

Comment:
This class looks pretty fine, but could the apply code from Settings be moved here, as that kinda looks into this object's privates so it would be nicer to move that into a method here and just call that from the settings.